packages/vuelidate/src/core.js

```diff
@@ -217,58 +209,55 @@ function createValidatorResult (rule, state, key) {
  */
 function createValidationResults (rules, state, key, parentKey) {
   // collect the property keys
-  const ruleKeys = Object.keys(rules)
-
   const $dirty = ref(false)
 
-  let result = {
+  const results = {
     $dirty,
     $touch: () => { $dirty.value = true },
     $reset: () => { $dirty.value = false }
   }
 
-  /**
-   * If there are no validation rules, it is most likely
-   * a top level state, aka root
-   */
-  if (!ruleKeys.length) return result
+  watch(rules, (changedRules) => {
+    /**
+     * If there are no validation rules, it is most likely
+     * a top level state, aka root
+     */
+    const ruleKeys = Object.keys(changedRules)
+    // if (!ruleKeys.length) return result
```

Comment:
I think the above comment targets this commented line. Do you know if we can early return here? If so, it might be a good idea to skip the following operations (even though on an empty array it would be pretty straightforward. Though we would need to add the `$invalid`, `$pending` and other keys to the `results` object.

Reply:
We can yeah. I had it at one point, but forgot to re add it.
We fallback to false in the extractMeta method.